src/memray/_memray/tracking_api.cpp

```diff
@@ -441,16 +480,43 @@ Tracker::childFork()
         return;
     }
 
+    assert(old_tracker != nullptr);
+
     // Re-enable tracking with a brand new tracker.
     d_instance_owner.reset(new Tracker(
             std::move(new_writer),
             old_tracker->d_unwind_native_frames,
             old_tracker->d_memory_interval,
             old_tracker->d_follow_fork,
-            old_tracker->d_trace_python_allocators));
+            old_tracker->d_trace_python_allocators,
+            old_tracker->d_sampling_interval));
     RecursionGuard::isActive = false;
 }
 
+// static std::unordered_set<void*> pointers;
+
+static constexpr char TRACE_MARK = 42;
+
+static inline void
+markPtr(void* ptr, const hooks::Allocator& func)
+{
+    if (func == hooks::Allocator::MMAP || isPymalloc(func)) {
+        return;
+    }
+    size_t usable_size = malloc_usable_size(ptr);
```

Comment:
In order for this to be safe, we need to start hooking `malloc_usable_size` and subtracting out our own overhead in the case where the mark is set, so that the user doesn't think it's safe to write into our extra buffer. Something like:
```cpp
size_t
malloc_usable_size(void* addr) noexcept
{
    assert(hooks::malloc_usable_size);
    size_t ret = hooks::malloc_usable_size(addr);
    if (ret > 0 && isPtrMarked(addr)) {
        ret -= MEMRAY_ALLOC_OVERHEAD;
    }
    return ret;
}
```
though that glosses over the fact that this stuff is spread across different translation units...